src/features/game/Game.tsx

```diff
@@ -37,6 +37,10 @@ import { screenTracker } from "lib/utils/screen";
 import { Withdrawn } from "./components/Withdrawn";
 import { EasterEggHunt } from "features/easter/Area";
 import { Resetting } from "features/auth/components/Resetting";
+import Decimal from 'decimal.js-light';
+
+const DECIMAL_PRECISION = 78;
```

Comment:
What is the thought process behind 78?

Could we use 18 decimal points to support ERC20 & other smart contract precision?

Reply:
According to the docs, the [precision ](https://mikemcl.github.io/decimal.js/#precision)setting in decimal.js is actually used to store the number of significant digits.

In the bug, the number is "1438.3999...99":   4 digits before point and 18 decimals --> 22 significant digits

ERC20 tokens usually have 18 decimals places, but it's up to the contract to use any number arbitrarily.

As smart contracts only use uint256, the maximum value that can be represented (2^256 - 1) can have at most 78 significant digits.

Although 78 is the most extreme case, I would say that 18 and "some more" would be enough in practice.

Reply:
Good point, thanks!